docs/gen_api_ref.py

```diff
@@ -0,0 +1,40 @@
+"""Generate the API reference pages and mkdocs nav."""
+from pathlib import Path
+
+import mkdocs_gen_files
+
+# flax.linen.Module gives mkdocstrings problems, so we filter out anything
+# that explicitly subclasses it, as well as private methods (which is default)
+FILTER_OUT_MODULE = """    handler: python
+    selection:
+        filters:
+        - "!^_[^_]"
+        - "!Module"
+"""
+
+if __name__ == "<run_path>":
+    nav = mkdocs_gen_files.Nav()
+
+    # leave out __init__.py files
+    package_paths = Path("vmcnet").glob("**/[!__init__]*.py")
+
+    for path in sorted(package_paths):
+        module_path = path.relative_to("vmcnet").with_suffix("")
+        doc_path = path.relative_to("vmcnet").with_suffix(".md")
+        full_doc_path = Path("api", doc_path)  # get path of file to be created
+
+        parts = list(module_path.parts)
+        nav[parts] = doc_path  # append to navigation
+
+        with mkdocs_gen_files.open(full_doc_path, "w") as fd:
+            # put in mkdocstrings magic
```

Comment:
How does this mkdocstrings magic work? I'm sure it's just my lack of understanding, but I don't really see where the key piece here of actually generating the docs happens.

Reply:
Basically the idea is to include stuff like `::: vmcnet.models.construct` in a markdown file and the mkdocstrings extension for mkdocs will grab all the classes and functions from that module and populate the resulting html file.